metabase/metabase
# 3. Update the select UX to match new design

Epic: https://github.com/metabase/metabase/issues/39246

### Description

This PR implements [the new design](https://www.figma.com/file/yOksAHQ5ih6ospZd6f3RVc/Let-white-labeling-customers-customize-illustrations?type=design&node-id=1486-122158&mode=design&t=ieKLYahwWumPT4eu-0) that has been changed since I merged the first PR for this project.

### How to verify

Go to any illustration settings
1. Starting from 0, when selecting "Custom" option, nothing should be saved, but it should show the UI to upload a custom image.
2. Click "Choose File" should popup the native file browser that only accepts SVG, JPG, and PNG files
3. If you upload a malformed image e.g. a PDF renamed as PNG. You should see an error message.
4. If you upload an image bigger than 2MB you should see an error message
5. If you upload a valid image with a size < 2MB we should see that the changes saved notification appears and the preview image of your selected image shown, alongside the file name.
6. If you navigate away and come back, the file name should disappear because we don't store file name anywhere. You should see `Remove uploaded image` label instead.
7. You can Click "Choose File" again to replace the uploaded image
8. You can click the X icon to remove the uploaded file, this should reset the value back to the default value whatever it used to be either Lighthouse or Sailboat.

### Demo

![image](https://github.com/metabase/metabase/assets/1937582/08487abc-c649-405e-83b2-bb1e14b72f79)

### Checklist

- [x] Tests have been added/updated to cover changes in this PR. (I only adjust the tests to pass with the new behavior, more tests will be added once I connect the new settings to places where we show those illustrations)
